src/PowerShellTips/2023-09-22-avoid-array-addition.ps1

```diff
@@ -0,0 +1,31 @@
+$tip = [tiPS.PowerShellTip]::new()
+$tip.CreatedDate = [DateTime]::Parse('2023-09-22')
+$tip.Title = 'Avoid Array addition'
+$tip.TipText = @'
+Array addition is a expensive and inefficient operation and can, most of the times, be replaced by PowerShell explicit loop assignment.
+
+Use a `List<T>` instead in those cases when adding to a collection while looping is required.
+'@
+$tip.Example = @'
+# Array addition:
+$result = @()
+foreach ($i in 0..10) {
+    $result += $i
+}
+
+# Can be easily replaced with explicit assignemnt:
+$result = foreach ($i in 0..10) {
+    $i
+}
+
+# And, when not possible, a List<T> is recommended:
+$result = [System.Collections.Generic.List[int]]::new()
+foreach ($i in 0..10) {
+    $result.Add($i)
+}
+'@
+$tip.Urls = @(
+	'https://learn.microsoft.com/en-us/powershell/scripting/dev-cross-plat/performance/script-authoring-considerations?view=powershell-7.3#array-addition'
+)
+$tip.MinPowerShellVersion = '5.1'
```

Comment:
Would this syntax not work in Windows PowerShell 3.0? Was something explicitly added in 5.1 for this? If this works in earlier versions of PowerShell, then I would leave this min version as `0.0` indicating it works in all PowerShell versions.

Reply:
Hi @deadlydog, tbh I wouldn't know, I've never used PowerShell 3.0. What I know for sure is that the `::new()` in my tip example wouldn't for 3.0 folks as that member was added in 5.0: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_intrinsic_members?view=powershell-7.3#new-method-for-types // folks in 4.0 (and maybe 3.0 (?) need to use the `New-Object` syntax). Good module btw, I like it 😉

Reply:
Speaking of `New-Object` that would be a good tip to add in the performance category 😅

Reply:
Ah, that makes sense. Your example code requires PS v5, but the tip advice itself does not 👍